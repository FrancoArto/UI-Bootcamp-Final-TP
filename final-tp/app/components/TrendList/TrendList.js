import React, { Component } from 'react';
import { View, Text } from 'native-base';
import styles from './trendList.style';
import { FlatList, ActivityIndicator } from 'react-native';
import Trend from '../Trend/Trend';

export default class TrendList extends Component {
  constructor(props) {
    super(props);

    this.renderItem = this.renderItem.bind(this);
    this.handleOnPress = this.handleOnPress.bind(this);
  }

  handleOnPress(event) {
    this.props.handleOnTrendPress(event);
  }

  renderItem({item, index}) {
    return (
        <View style={styles.row}>
            <Trend onPress={this.handleOnPress} index={index} item={item} />
        </View>
    )
}


  render() {
    if (this.props.loading) {
      return (
          <View style={styles.activityIndicatorContainer}>
              <ActivityIndicator animating={true}/>
          </View>
      );
    } else {
      return (
          <View style={styles.container}>
              <Text>Trends for you</Text>
              <FlatList
                  ref='trendList'
                  data={this.props.data}
                  renderItem={this.renderItem}
                  keyExtractor={(item, index) => index.toString()}/>
          </View>
      );
    }
  }
}