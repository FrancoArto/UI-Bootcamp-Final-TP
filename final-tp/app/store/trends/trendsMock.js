export const trendsArray = [
  {
    "trends": [
      {
        "name": "Nahir Galarza",
        "url": "http://twitter.com/search?q=%22Nahir+Galarza%22",
        "promoted_content": null,
        "query": "%22Nahir+Galarza%22",
        "tweet_volume": null
      },
      {
        "name": "#FiestOT18",
        "url": "http://twitter.com/search?q=%23FiestOT18",
        "promoted_content": null,
        "query": "%23FiestOT18",
        "tweet_volume": null
      },
      {
        "name": "#NoMeMolestas",
        "url": "http://twitter.com/search?q=%23NoMeMolestas",
        "promoted_content": null,
        "query": "%23NoMeMolestas",
        "tweet_volume": null
      },
      {
        "name": "Mandzukic",
        "url": "http://twitter.com/search?q=Mandzukic",
        "promoted_content": null,
        "query": "Mandzukic",
        "tweet_volume": null
      },
      {
        "name": "#Ligue1xESPN",
        "url": "http://twitter.com/search?q=%23Ligue1xESPN",
        "promoted_content": null,
        "query": "%23Ligue1xESPN",
        "tweet_volume": null
      },
      {
        "name": "#LoMejorDeCombate",
        "url": "http://twitter.com/search?q=%23LoMejorDeCombate",
        "promoted_content": null,
        "query": "%23LoMejorDeCombate",
        "tweet_volume": null
      },
      {
        "name": "Rosedal",
        "url": "http://twitter.com/search?q=Rosedal",
        "promoted_content": null,
        "query": "Rosedal",
        "tweet_volume": null
      },
      {
        "name": "La Educación Pública",
        "url": "http://twitter.com/search?q=%22La+Educaci%C3%B3n+P%C3%BAblica%22",
        "promoted_content": null,
        "query": "%22La+Educaci%C3%B3n+P%C3%BAblica%22",
        "tweet_volume": null
      },
      {
        "name": "Aduriz",
        "url": "http://twitter.com/search?q=Aduriz",
        "promoted_content": null,
        "query": "Aduriz",
        "tweet_volume": 15583
      },
      {
        "name": "Bidoglio",
        "url": "http://twitter.com/search?q=Bidoglio",
        "promoted_content": null,
        "query": "Bidoglio",
        "tweet_volume": null
      },
      {
        "name": "La Angostura",
        "url": "http://twitter.com/search?q=%22La+Angostura%22",
        "promoted_content": null,
        "query": "%22La+Angostura%22",
        "tweet_volume": null
      },
      {
        "name": "#クリスマスボックス",
        "url": "http://twitter.com/search?q=%23%E3%82%AF%E3%83%AA%E3%82%B9%E3%83%9E%E3%82%B9%E3%83%9C%E3%83%83%E3%82%AF%E3%82%B9",
        "promoted_content": null,
        "query": "%23%E3%82%AF%E3%83%AA%E3%82%B9%E3%83%9E%E3%82%B9%E3%83%9C%E3%83%83%E3%82%AF%E3%82%B9",
        "tweet_volume": 2494801
      },
      {
        "name": "Rafael Santos Borré",
        "url": "http://twitter.com/search?q=%22Rafael+Santos+Borr%C3%A9%22",
        "promoted_content": null,
        "query": "%22Rafael+Santos+Borr%C3%A9%22",
        "tweet_volume": null
      },
      {
        "name": "El Real Madrid",
        "url": "http://twitter.com/search?q=%22El+Real+Madrid%22",
        "promoted_content": null,
        "query": "%22El+Real+Madrid%22",
        "tweet_volume": 23662
      },
      {
        "name": "Lionel",
        "url": "http://twitter.com/search?q=Lionel",
        "promoted_content": null,
        "query": "Lionel",
        "tweet_volume": 21320
      },
      {
        "name": "Sugar",
        "url": "http://twitter.com/search?q=Sugar",
        "promoted_content": null,
        "query": "Sugar",
        "tweet_volume": 112213
      },
      {
        "name": "Pity",
        "url": "http://twitter.com/search?q=Pity",
        "promoted_content": null,
        "query": "Pity",
        "tweet_volume": 79391
      },
      {
        "name": "Alfaro",
        "url": "http://twitter.com/search?q=Alfaro",
        "promoted_content": null,
        "query": "Alfaro",
        "tweet_volume": 55308
      },
      {
        "name": "Navidad",
        "url": "http://twitter.com/search?q=Navidad",
        "promoted_content": null,
        "query": "Navidad",
        "tweet_volume": 557910
      },
      {
        "name": "Papá Noel",
        "url": "http://twitter.com/search?q=%22Pap%C3%A1+Noel%22",
        "promoted_content": null,
        "query": "%22Pap%C3%A1+Noel%22",
        "tweet_volume": 49989
      },
      {
        "name": "De la Cruz",
        "url": "http://twitter.com/search?q=%22De+la+Cruz%22",
        "promoted_content": null,
        "query": "%22De+la+Cruz%22",
        "tweet_volume": null
      },
      {
        "name": "álvarez",
        "url": "http://twitter.com/search?q=%C3%A1lvarez",
        "promoted_content": null,
        "query": "%C3%A1lvarez",
        "tweet_volume": 14224
      },
      {
        "name": "Zuculini",
        "url": "http://twitter.com/search?q=Zuculini",
        "promoted_content": null,
        "query": "Zuculini",
        "tweet_volume": null
      },
      {
        "name": "Arroyo Salgado",
        "url": "http://twitter.com/search?q=%22Arroyo+Salgado%22",
        "promoted_content": null,
        "query": "%22Arroyo+Salgado%22",
        "tweet_volume": 16473
      },
      {
        "name": "Bird Box",
        "url": "http://twitter.com/search?q=%22Bird+Box%22",
        "promoted_content": null,
        "query": "%22Bird+Box%22",
        "tweet_volume": 81792
      },
      {
        "name": "Nelson Castro",
        "url": "http://twitter.com/search?q=%22Nelson+Castro%22",
        "promoted_content": null,
        "query": "%22Nelson+Castro%22",
        "tweet_volume": null
      },
      {
        "name": "Felices Fiestas",
        "url": "http://twitter.com/search?q=%22Felices+Fiestas%22",
        "promoted_content": null,
        "query": "%22Felices+Fiestas%22",
        "tweet_volume": 52778
      },
      {
        "name": "Gonzalo Martínez",
        "url": "http://twitter.com/search?q=%22Gonzalo+Mart%C3%ADnez%22",
        "promoted_content": null,
        "query": "%22Gonzalo+Mart%C3%ADnez%22",
        "tweet_volume": null
      },
      {
        "name": "Luca Prodan",
        "url": "http://twitter.com/search?q=%22Luca+Prodan%22",
        "promoted_content": null,
        "query": "%22Luca+Prodan%22",
        "tweet_volume": null
      },
      {
        "name": "Sergio Ramos",
        "url": "http://twitter.com/search?q=%22Sergio+Ramos%22",
        "promoted_content": null,
        "query": "%22Sergio+Ramos%22",
        "tweet_volume": 23106
      },
      {
        "name": "Juan Grabois",
        "url": "http://twitter.com/search?q=%22Juan+Grabois%22",
        "promoted_content": null,
        "query": "%22Juan+Grabois%22",
        "tweet_volume": null
      },
      {
        "name": "Vamos River",
        "url": "http://twitter.com/search?q=%22Vamos+River%22",
        "promoted_content": null,
        "query": "%22Vamos+River%22",
        "tweet_volume": null
      },
      {
        "name": "Federico Moura",
        "url": "http://twitter.com/search?q=%22Federico+Moura%22",
        "promoted_content": null,
        "query": "%22Federico+Moura%22",
        "tweet_volume": null
      },
      {
        "name": "Nisman",
        "url": "http://twitter.com/search?q=Nisman",
        "promoted_content": null,
        "query": "Nisman",
        "tweet_volume": 21963
      },
      {
        "name": "Crystal Palace",
        "url": "http://twitter.com/search?q=%22Crystal+Palace%22",
        "promoted_content": null,
        "query": "%22Crystal+Palace%22",
        "tweet_volume": 53087
      },
      {
        "name": "Facundo Daelli",
        "url": "http://twitter.com/search?q=%22Facundo+Daelli%22",
        "promoted_content": null,
        "query": "%22Facundo+Daelli%22",
        "tweet_volume": null
      },
      {
        "name": "Santiago del Moro",
        "url": "http://twitter.com/search?q=%22Santiago+del+Moro%22",
        "promoted_content": null,
        "query": "%22Santiago+del+Moro%22",
        "tweet_volume": null
      },
      {
        "name": "Liverpool",
        "url": "http://twitter.com/search?q=Liverpool",
        "promoted_content": null,
        "query": "Liverpool",
        "tweet_volume": 141820
      },
      {
        "name": "Celta",
        "url": "http://twitter.com/search?q=Celta",
        "promoted_content": null,
        "query": "Celta",
        "tweet_volume": 22292
      },
      {
        "name": "Nochebuena",
        "url": "http://twitter.com/search?q=Nochebuena",
        "promoted_content": null,
        "query": "Nochebuena",
        "tweet_volume": 12987
      },
      {
        "name": "Abu Dhabi",
        "url": "http://twitter.com/search?q=%22Abu+Dhabi%22",
        "promoted_content": null,
        "query": "%22Abu+Dhabi%22",
        "tweet_volume": 12414
      },
      {
        "name": "#BuenSabado",
        "url": "http://twitter.com/search?q=%23BuenSabado",
        "promoted_content": null,
        "query": "%23BuenSabado",
        "tweet_volume": 15221
      },
      {
        "name": "#serrandincampeon",
        "url": "http://twitter.com/search?q=%23serrandincampeon",
        "promoted_content": null,
        "query": "%23serrandincampeon",
        "tweet_volume": null
      },
      {
        "name": "#PeligroSinCodificar",
        "url": "http://twitter.com/search?q=%23PeligroSinCodificar",
        "promoted_content": null,
        "query": "%23PeligroSinCodificar",
        "tweet_volume": null
      },
      {
        "name": "#ElDementeIsOverParty",
        "url": "http://twitter.com/search?q=%23ElDementeIsOverParty",
        "promoted_content": null,
        "query": "%23ElDementeIsOverParty",
        "tweet_volume": null
      },
      {
        "name": "#PongamosUnaFotoDeCFK",
        "url": "http://twitter.com/search?q=%23PongamosUnaFotoDeCFK",
        "promoted_content": null,
        "query": "%23PongamosUnaFotoDeCFK",
        "tweet_volume": null
      },
      {
        "name": "#SimonCowellOverParty",
        "url": "http://twitter.com/search?q=%23SimonCowellOverParty",
        "promoted_content": null,
        "query": "%23SimonCowellOverParty",
        "tweet_volume": 261166
      },
      {
        "name": "#LaVozElTeatro",
        "url": "http://twitter.com/search?q=%23LaVozElTeatro",
        "promoted_content": null,
        "query": "%23LaVozElTeatro",
        "tweet_volume": null
      },
      {
        "name": "#FelizAñoNuevo",
        "url": "http://twitter.com/search?q=%23FelizA%C3%B1oNuevo",
        "promoted_content": null,
        "query": "%23FelizA%C3%B1oNuevo",
        "tweet_volume": null
      },
      {
        "name": "#Super20",
        "url": "http://twitter.com/search?q=%23Super20",
        "promoted_content": null,
        "query": "%23Super20",
        "tweet_volume": null
      }
    ],
    "as_of": "2018-12-27T19:25:50Z",
    "created_at": "2018-12-22T23:47:53Z",
    "locations": [
      {
        "name": "Argentina",
        "woeid": 23424747
      }
    ]
  }
]