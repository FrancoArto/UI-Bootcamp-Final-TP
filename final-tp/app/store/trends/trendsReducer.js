
import { FETCH_TRENDS_BEGIN, FETCH_TRENDS_SUCCESS, FETCH_TRENDS_FAILURE} from './trendsActions'
import { trendsInitialState } from '../initialState'
 
const trendsReducer = (state = trendsInitialState, action) => {
    switch (action.type) {
        case FETCH_TRENDS_BEGIN:{
            return {
                ...state,
                 loading:true,
                 error: null,
              }
        }           
        case FETCH_TRENDS_SUCCESS:{
            return {
                ...state,
                loading: false,
                error: null,
                data: action.payload,
            }
        }
        case FETCH_TRENDS_FAILURE:{
            return {
                ...state,
                loading: false,
                error: action.payload,
                data: [],
            }
        }                
        default:
            return state;
    }
};
export default trendsReducer;