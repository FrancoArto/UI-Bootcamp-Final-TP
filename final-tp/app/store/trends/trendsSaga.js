import { FETCH_TRENDS_BEGIN, fetchTrendsSuccess, fetchTrendsError } from './trendsActions';
import { takeLatest, put, call } from 'redux-saga/effects';
import { GET_ARGTRENDS_URL } from '../../api/apiUrls';

export function* trendsSaga() {
  yield takeLatest(FETCH_TRENDS_BEGIN, fetchTrends)  
}

export function* fetchTrends() {
  try {
    const response = yield call(fetch, GET_ARGTRENDS_URL())
    const data = yield call([response, "json"]);
    yield put(fetchTrendsSuccess(data[0].trends))
  } catch (er) {
    yield put(fetchTrendsError(er))
  }  
}