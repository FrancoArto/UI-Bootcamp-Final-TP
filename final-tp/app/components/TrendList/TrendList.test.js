import React from 'react';
import {mount, shallow} from 'enzyme'
import { trendsArray } from '../../store/trends/trendsMock';
import TrendList from './TrendList';


const trends = trendsArray


it('renders without crashing', () => {
  const rendered = shallow(<TrendList data={trends[0].trends}/>)
  expect(rendered).toMatchSnapshot();
});

it('renders activity indicator without crashing', () => {
  const rendered = shallow(<TrendList loading={true}/>)
  expect(rendered).toMatchSnapshot();
});

it('handles trend press', () => {
  const onPress =jest.fn()
  const wrapper = mount(<TrendList data={trends[0].trends} handleOnTrendPress={onPress}/>)

  const trend = wrapper.find('Trend').first();
  trend.props().onPress();
  expect(onPress).toHaveBeenCalled()
});