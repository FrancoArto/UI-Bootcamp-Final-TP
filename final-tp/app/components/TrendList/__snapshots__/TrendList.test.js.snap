// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders activity indicator without crashing 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "alignItems": "center",
      "flex": 0.85,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</Styled(ViewNB)>
`;

exports[`renders without crashing 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "flex": 0.85,
      "paddingLeft": 10,
    }
  }
>
  <Styled(Text)>
    Trends for you
  </Styled(Text)>
  <FlatList
    data={
      Array [
        Object {
          "name": "Nahir Galarza",
          "promoted_content": null,
          "query": "%22Nahir+Galarza%22",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%22Nahir+Galarza%22",
        },
        Object {
          "name": "#FiestOT18",
          "promoted_content": null,
          "query": "%23FiestOT18",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%23FiestOT18",
        },
        Object {
          "name": "#NoMeMolestas",
          "promoted_content": null,
          "query": "%23NoMeMolestas",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%23NoMeMolestas",
        },
        Object {
          "name": "Mandzukic",
          "promoted_content": null,
          "query": "Mandzukic",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=Mandzukic",
        },
        Object {
          "name": "#Ligue1xESPN",
          "promoted_content": null,
          "query": "%23Ligue1xESPN",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%23Ligue1xESPN",
        },
        Object {
          "name": "#LoMejorDeCombate",
          "promoted_content": null,
          "query": "%23LoMejorDeCombate",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%23LoMejorDeCombate",
        },
        Object {
          "name": "Rosedal",
          "promoted_content": null,
          "query": "Rosedal",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=Rosedal",
        },
        Object {
          "name": "La Educación Pública",
          "promoted_content": null,
          "query": "%22La+Educaci%C3%B3n+P%C3%BAblica%22",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%22La+Educaci%C3%B3n+P%C3%BAblica%22",
        },
        Object {
          "name": "Aduriz",
          "promoted_content": null,
          "query": "Aduriz",
          "tweet_volume": 15583,
          "url": "http://twitter.com/search?q=Aduriz",
        },
        Object {
          "name": "Bidoglio",
          "promoted_content": null,
          "query": "Bidoglio",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=Bidoglio",
        },
        Object {
          "name": "La Angostura",
          "promoted_content": null,
          "query": "%22La+Angostura%22",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%22La+Angostura%22",
        },
        Object {
          "name": "#クリスマスボックス",
          "promoted_content": null,
          "query": "%23%E3%82%AF%E3%83%AA%E3%82%B9%E3%83%9E%E3%82%B9%E3%83%9C%E3%83%83%E3%82%AF%E3%82%B9",
          "tweet_volume": 2494801,
          "url": "http://twitter.com/search?q=%23%E3%82%AF%E3%83%AA%E3%82%B9%E3%83%9E%E3%82%B9%E3%83%9C%E3%83%83%E3%82%AF%E3%82%B9",
        },
        Object {
          "name": "Rafael Santos Borré",
          "promoted_content": null,
          "query": "%22Rafael+Santos+Borr%C3%A9%22",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%22Rafael+Santos+Borr%C3%A9%22",
        },
        Object {
          "name": "El Real Madrid",
          "promoted_content": null,
          "query": "%22El+Real+Madrid%22",
          "tweet_volume": 23662,
          "url": "http://twitter.com/search?q=%22El+Real+Madrid%22",
        },
        Object {
          "name": "Lionel",
          "promoted_content": null,
          "query": "Lionel",
          "tweet_volume": 21320,
          "url": "http://twitter.com/search?q=Lionel",
        },
        Object {
          "name": "Sugar",
          "promoted_content": null,
          "query": "Sugar",
          "tweet_volume": 112213,
          "url": "http://twitter.com/search?q=Sugar",
        },
        Object {
          "name": "Pity",
          "promoted_content": null,
          "query": "Pity",
          "tweet_volume": 79391,
          "url": "http://twitter.com/search?q=Pity",
        },
        Object {
          "name": "Alfaro",
          "promoted_content": null,
          "query": "Alfaro",
          "tweet_volume": 55308,
          "url": "http://twitter.com/search?q=Alfaro",
        },
        Object {
          "name": "Navidad",
          "promoted_content": null,
          "query": "Navidad",
          "tweet_volume": 557910,
          "url": "http://twitter.com/search?q=Navidad",
        },
        Object {
          "name": "Papá Noel",
          "promoted_content": null,
          "query": "%22Pap%C3%A1+Noel%22",
          "tweet_volume": 49989,
          "url": "http://twitter.com/search?q=%22Pap%C3%A1+Noel%22",
        },
        Object {
          "name": "De la Cruz",
          "promoted_content": null,
          "query": "%22De+la+Cruz%22",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%22De+la+Cruz%22",
        },
        Object {
          "name": "álvarez",
          "promoted_content": null,
          "query": "%C3%A1lvarez",
          "tweet_volume": 14224,
          "url": "http://twitter.com/search?q=%C3%A1lvarez",
        },
        Object {
          "name": "Zuculini",
          "promoted_content": null,
          "query": "Zuculini",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=Zuculini",
        },
        Object {
          "name": "Arroyo Salgado",
          "promoted_content": null,
          "query": "%22Arroyo+Salgado%22",
          "tweet_volume": 16473,
          "url": "http://twitter.com/search?q=%22Arroyo+Salgado%22",
        },
        Object {
          "name": "Bird Box",
          "promoted_content": null,
          "query": "%22Bird+Box%22",
          "tweet_volume": 81792,
          "url": "http://twitter.com/search?q=%22Bird+Box%22",
        },
        Object {
          "name": "Nelson Castro",
          "promoted_content": null,
          "query": "%22Nelson+Castro%22",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%22Nelson+Castro%22",
        },
        Object {
          "name": "Felices Fiestas",
          "promoted_content": null,
          "query": "%22Felices+Fiestas%22",
          "tweet_volume": 52778,
          "url": "http://twitter.com/search?q=%22Felices+Fiestas%22",
        },
        Object {
          "name": "Gonzalo Martínez",
          "promoted_content": null,
          "query": "%22Gonzalo+Mart%C3%ADnez%22",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%22Gonzalo+Mart%C3%ADnez%22",
        },
        Object {
          "name": "Luca Prodan",
          "promoted_content": null,
          "query": "%22Luca+Prodan%22",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%22Luca+Prodan%22",
        },
        Object {
          "name": "Sergio Ramos",
          "promoted_content": null,
          "query": "%22Sergio+Ramos%22",
          "tweet_volume": 23106,
          "url": "http://twitter.com/search?q=%22Sergio+Ramos%22",
        },
        Object {
          "name": "Juan Grabois",
          "promoted_content": null,
          "query": "%22Juan+Grabois%22",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%22Juan+Grabois%22",
        },
        Object {
          "name": "Vamos River",
          "promoted_content": null,
          "query": "%22Vamos+River%22",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%22Vamos+River%22",
        },
        Object {
          "name": "Federico Moura",
          "promoted_content": null,
          "query": "%22Federico+Moura%22",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%22Federico+Moura%22",
        },
        Object {
          "name": "Nisman",
          "promoted_content": null,
          "query": "Nisman",
          "tweet_volume": 21963,
          "url": "http://twitter.com/search?q=Nisman",
        },
        Object {
          "name": "Crystal Palace",
          "promoted_content": null,
          "query": "%22Crystal+Palace%22",
          "tweet_volume": 53087,
          "url": "http://twitter.com/search?q=%22Crystal+Palace%22",
        },
        Object {
          "name": "Facundo Daelli",
          "promoted_content": null,
          "query": "%22Facundo+Daelli%22",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%22Facundo+Daelli%22",
        },
        Object {
          "name": "Santiago del Moro",
          "promoted_content": null,
          "query": "%22Santiago+del+Moro%22",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%22Santiago+del+Moro%22",
        },
        Object {
          "name": "Liverpool",
          "promoted_content": null,
          "query": "Liverpool",
          "tweet_volume": 141820,
          "url": "http://twitter.com/search?q=Liverpool",
        },
        Object {
          "name": "Celta",
          "promoted_content": null,
          "query": "Celta",
          "tweet_volume": 22292,
          "url": "http://twitter.com/search?q=Celta",
        },
        Object {
          "name": "Nochebuena",
          "promoted_content": null,
          "query": "Nochebuena",
          "tweet_volume": 12987,
          "url": "http://twitter.com/search?q=Nochebuena",
        },
        Object {
          "name": "Abu Dhabi",
          "promoted_content": null,
          "query": "%22Abu+Dhabi%22",
          "tweet_volume": 12414,
          "url": "http://twitter.com/search?q=%22Abu+Dhabi%22",
        },
        Object {
          "name": "#BuenSabado",
          "promoted_content": null,
          "query": "%23BuenSabado",
          "tweet_volume": 15221,
          "url": "http://twitter.com/search?q=%23BuenSabado",
        },
        Object {
          "name": "#serrandincampeon",
          "promoted_content": null,
          "query": "%23serrandincampeon",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%23serrandincampeon",
        },
        Object {
          "name": "#PeligroSinCodificar",
          "promoted_content": null,
          "query": "%23PeligroSinCodificar",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%23PeligroSinCodificar",
        },
        Object {
          "name": "#ElDementeIsOverParty",
          "promoted_content": null,
          "query": "%23ElDementeIsOverParty",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%23ElDementeIsOverParty",
        },
        Object {
          "name": "#PongamosUnaFotoDeCFK",
          "promoted_content": null,
          "query": "%23PongamosUnaFotoDeCFK",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%23PongamosUnaFotoDeCFK",
        },
        Object {
          "name": "#SimonCowellOverParty",
          "promoted_content": null,
          "query": "%23SimonCowellOverParty",
          "tweet_volume": 261166,
          "url": "http://twitter.com/search?q=%23SimonCowellOverParty",
        },
        Object {
          "name": "#LaVozElTeatro",
          "promoted_content": null,
          "query": "%23LaVozElTeatro",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%23LaVozElTeatro",
        },
        Object {
          "name": "#FelizAñoNuevo",
          "promoted_content": null,
          "query": "%23FelizA%C3%B1oNuevo",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%23FelizA%C3%B1oNuevo",
        },
        Object {
          "name": "#Super20",
          "promoted_content": null,
          "query": "%23Super20",
          "tweet_volume": null,
          "url": "http://twitter.com/search?q=%23Super20",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</Styled(ViewNB)>
`;
